pkg/cmd/server/logs.go

```diff
@@ -42,66 +57,137 @@ var logsCmd = &cobra.Command{
 		}
 
 		switch {
-		case fileFlag:
-			return readServerLogFile()
-
+		case localFlag:
+			return readLocalServerLogFile()
 		default:
-			ws, res, err := apiclient.GetWebsocketConn(context.Background(), "/log/server", &activeProfile, &query)
+			return readRemoteServerLogFile(ctx, activeProfile, query)
+		}
+	},
+}
 
-			if err != nil {
-				log.Error(apiclient.HandleErrorResponse(res, err))
+func readRemoteServerLogFile(ctx context.Context, activeProfile config.Profile, query string) error {
+	apiclient, err := apiclient_util.GetApiClient(&activeProfile)
+	if err != nil {
+		return err
+	}
 
-				if activeProfile.Id != "default" {
-					return nil
-				}
+	logFiles, res, err := apiclient.ServerAPI.GetServerLogFiles(ctx).Execute()
+	if err != nil {
+		return apiclient_util.HandleErrorResponse(res, err)
+	}
 
-				readLogsFile := true
-				form := huh.NewForm(
-					huh.NewGroup(
-						huh.NewConfirm().Description("An error occurred while connecting to the server. Would you like to read from local log file instead?").
-							Value(&readLogsFile),
-					),
-				).WithTheme(views.GetCustomTheme())
-				formErr := form.Run()
-				if formErr != nil {
-					return formErr
-				}
+	if len(logFiles) == 0 {
+		return promptReadingLocalServerLogs("No server log files found.")
+	}
 
-				if readLogsFile {
-					return readServerLogFile()
-				}
-				return nil
+	if fileFlag == "" && len(logFiles) == 1 {
+		fileFlag = logFiles[0]
+	}
 
-			}
+	if fileFlag == "" {
+		selectedFile := selection.GetLogFileFromPrompt(logFiles)
+		if selectedFile == nil {
+			return nil
+		}
 
-			for {
-				_, msg, err := ws.ReadMessage()
-				if err != nil {
-					return nil
-				}
+		fileFlag = *selectedFile
+	}
 
-				fmt.Println(string(msg))
-			}
+	if fileFlag != "" {
```

Comment:
![Screenshot 2024-11-21 at 14 29 08](https://github.com/user-attachments/assets/6e7fde98-4008-41c0-89fa-0ac14b4f18f1)

This should throw 404.